.changeset/blue-eels-invite.md

```diff
@@ -0,0 +1,10 @@
+---
+"@fluidframework/tree": minor
+---
+---
+"section": tree
+---
+
+Recursive schema correctness is improved
```

Comment:
As this is a customer facing message, I don't think this is the right description. No existing schema are more correct.
 I think this would be better:

```suggestion
Better type errors for invalid recursive schema
```